File: app/src/main/java/com/simley/ndk_day78/player/video/VideoPlayerActivity.java
package com.simley.ndk_day78.player.video;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Environment;
import android.view.Surface;
import android.view.SurfaceHolder;

import com.simley.ndk_day78.R;
import com.simley.ndk_day78.databinding.ActivityVideoPlayerBinding;
import com.simley.ndk_day78.player.YEPlayer;

import java.io.File;

public class VideoPlayerActivity extends AppCompatActivity {

    private YEPlayer yePlayer;
    private ActivityVideoPlayerBinding binding;
    private Surface surface;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityVideoPlayerBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        yePlayer = new YEPlayer();

        binding.play.setOnClickListener(v -> {
            String mp4File = new File(Environment.getExternalStorageDirectory(), "/Movies/kali_hacker_video.mp4").getAbsolutePath();
            new Thread(() -> yePlayer.play(mp4File, surface)).start();
        });

        renderFrameToSurfaceView();
    }

    private void renderFrameToSurfaceView() {
        SurfaceHolder holder = binding.surfaceView.getHolder();
        holder.addCallback(new SurfaceHolder.Callback() {
            @Override
            public void surfaceCreated(@NonNull SurfaceHolder holder) {
                surface = holder.getSurface();
            }

            @Override
            public void surfaceChanged(@NonNull SurfaceHolder holder, int format, int width, int height) {

            }

            @Override
            public void surfaceDestroyed(@NonNull SurfaceHolder holder) {

            }
        });
    }

}

File: app/src/main/java/com/simley/ndk_day78/player/video/VideoPlayerActivity.kt
package com.simley.ndk_day78.player.video

import android.os.Bundle
import android.os.Environment
import android.view.LayoutInflater
import android.view.Surface
import android.view.SurfaceHolder
import android.view.View
import com.simley.lib_base.BaseActivity
import com.simley.ndk_day78.databinding.ActivityVideoPlayerBinding
import com.simley.ndk_day78.player.YEPlayer
import com.simley.ndk_day78.utils.ThreadPool
import java.io.File

class VideoPlayerActivity : BaseActivity<ActivityVideoPlayerBinding>(), View.OnClickListener {

    private val yePlayer: YEPlayer = YEPlayer()

    private var surface: Surface? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        binding.play.setOnClickListener(this)
        renderFrameToSurfaceView()
    }


    private fun renderFrameToSurfaceView() {
        val holder = binding.surfaceView.holder
        holder.addCallback(object : SurfaceHolder.Callback {
            override fun surfaceCreated(holder: SurfaceHolder) {
                surface = holder.surface
            }

            override fun surfaceChanged(
                holder: SurfaceHolder,
                format: Int,
                width: Int,
                height: Int
            ) {
            }

            override fun surfaceDestroyed(holder: SurfaceHolder) {}
        })
    }

    override fun onClick(p0: View?) {
        val mp4File = File(
            Environment.getExternalStorageDirectory(),
            "/Movies/kali_hacker_video.mp4"
        ).absolutePath
        ThreadPool.getInstance().execute { yePlayer.play(mp4File, surface) }
    }

    override fun getViewBinding(layoutInflater: LayoutInflater): ActivityVideoPlayerBinding =
        ActivityVideoPlayerBinding.inflate(layoutInflater, null, false)

}